### DISCUSSION TITLE
what is the difference between “ vimp, subsample, holdout.vimp”？

### DISCUSSION THREAD

**Discussion by A-Pai** [kogalur/randomForestSRC#141] [Q&A] [answered]
what is the difference between “ vimp, subsample, holdout.vimp”？

**Comment by ishwaran** [comment, marked as answer]
`vimp` is the function for computing variable importance (VIMP) and is used for feature selection.  An example and well known method for VIMP used by our package is Breiman-Cutler permutation VIMP.  How VIMP is calculated and background material is provided in the vignette https://luminwin.github.io/randomForestSRC/articles/rfsrc-subsample.html and its help file is here https://luminwin.github.io/randomForestSRC/reference/vimp.rfsrc.html

`subsample` is a function that uses resampling to calculate the standard error for VIMP and for constructing confidence intervals for VIMP.  You can read more about it in the same vignette listed above

`holdout.vimp` is a different type of VIMP.  Rather than using a noising up technique (like Breiman-Cutler permutation), holdout vimp literally does what it says: it holds out the variable and determines how much prediction error increases or decreases with the variable in or out of the model. Here is the help file describing it in more detail https://luminwin.github.io/randomForestSRC/reference/holdout.vimp.rfsrc.html

**Reply by A-Pai** [reply]
What are the advantages of Holdout VIMP compared with the noising up technique (like Breiman-Cutler permutation)？
